### PR TITLE
[MOV] im_livechat: move Thread outside of public_livechat.js

### PR DESCRIPTION
**Pull request by Nemael**
Task-2212347

### CODE REVIEW

File: addons/im_livechat/static/src/legacy/public_livechat.js [hunk 1]
```diff
@@ -2850,692 +2850,6 @@ return Timers;
...
-                displayAuthorMessages[message.getID()] = !options.squashCloseMessages;
-            }
-            prevMessage = message;
-        });
-
-        if (modeOptions.displayOrder === ORDER.DESC) {
-            messages.reverse();
-        }
-
-        this.$el.html(QWeb.render('im_livechat.legacy.mail.widget.Thread', {
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/legacy/public_livechat.js:3043]: ci/security due to this move

File: addons/im_livechat/static/src/legacy/public_livechat.js [hunk 2]
```diff
@@ -2850,692 +2850,6 @@ return Timers;
...
-            this._renderMessageSeenPopover(thread, messages);
-        }
-    },
-
-    /**
-     * Render thread widget when loading, i.e. when messaging is not yet ready.
-     * @see /mail/init_messaging
-     */
-    renderLoading: function () {
-        this.$el.html(QWeb.render('im_livechat.legacy.mail.widget.ThreadLoading'));
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/legacy/public_livechat.js:3079]: ci/security due to this move

File: addons/im_livechat/static/src/legacy/public_livechat.js [hunk 3]
```diff
@@ -2850,692 +2850,6 @@ return Timers;
...
-     * Modifies $element to add the 'read more/read less' functionality
-     * All element nodes with 'data-o-mail-quote' attribute are concerned.
-     * All text nodes after a ``#stopSpelling`` element are concerned.
-     * Those text nodes need to be wrapped in a span (toggle functionality).
-     * All consecutive elements are joined in one 'read more/read less'.
-     *
-     * @private
-     * @param {jQuery} $element
-     */
-    _insertReadMore: function ($element) {
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/legacy/public_livechat.js:3208]: maybe ci/security due to this function being moved?

File: addons/im_livechat/static/src/legacy/public_livechat.js [hunk 4]
```diff
@@ -2850,692 +2850,6 @@ return Timers;
...
-        });
-    },
-    /**
-     * @private
-     */
-    _updateTimestamps: function () {
-        var isAtBottom = this.isAtBottom();
-        this.$('.o_mail_timestamp').each(function () {
-            var date = $(this).data('date');
-            $(this).html(mailUtils.timeFromNow(date));
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/legacy/public_livechat.js:3387]: ci/security due to this move

**Review by alexkuhn — APPROVED** [review]
@seb-odoo ci/security due to set `html` LOCs being moved

File: addons/im_livechat/static/src/legacy/public_livechat.js (hunk 4 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/legacy/public_livechat.js:3387]: Looks wrong to use html in this case, timeFromNow is just a string, should set element text instead?

File: addons/im_livechat/static/src/legacy/public_livechat.js (hunk 2 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/legacy/public_livechat.js:3079]: This one is fine, especially the value is coming from a template.

File: addons/im_livechat/static/src/legacy/public_livechat.js (hunk 1 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/legacy/public_livechat.js:3043]: This one is fine, especially the value is coming from a template.

File: addons/im_livechat/static/src/legacy/public_livechat.js (hunk 3 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/legacy/public_livechat.js:3208]: I don't see any flag at this line.

File: addons/im_livechat/static/src/legacy/public_livechat.js (hunk 4 above)
Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/public_livechat.js:3387]: Indeed it's strange to use `html`, but keep in mind PR is just a move.

Comment (seb-odoo) [reply addons/im_livechat/static/src/legacy/public_livechat.js:3387]: Fair enough, but should make a prep PR to keep minimal diff here then, not the other way around.

**Comment by seb-odoo** [comment]
@robodoo override=ci/security
Legacy code move still requiring to manually set html. Content is safe as it is coming from a template.

**Comment by seb-odoo** [comment]
obsolete pr

**Pull request closed without merging**